File: examples/dotnet/Clprolf.Examples.Graphs/Clprolf.Examples.Graphs/Worker/Impl/GraphWorkerImpl.cs
namespace Clprolf.Examples.Graphs.Worker.Impl
{
  
    using global::Clprolf.ArchUnitNet.Attributes;
    using global::Clprolf.Examples.Graph.Workers;
    using global::Clprolf.Examples.Graphs.Agents.Org.Clprolf.Graph.Agents;
  
    using System;
    using System.Collections.Generic;

    namespace Clprolf.Examples.Graphs.Worker.Impl
    {
        [ClWorker]
        public class GraphWorkerImpl : IGraphWorker
        {
            private readonly IGraph _graph;

            // Le constructeur reçoit l'implémentation concrète (GraphImpl) comme en Java
            public GraphWorkerImpl(IGraph theGraph)
            {
                _graph = theGraph;
            }

            // Display graph structure
            public void PrintGraph()
            {
                var adj = _graph.Adj;

                // En C#, on itère directement sur le dictionnaire (pas besoin de entrySet)
                foreach (var entry in adj)
                {
                    // string.Join permet d'afficher proprement le contenu de la List<int>
                    Console.WriteLine($"{entry.Key} -> [{string.Join(", ", entry.Value)}]");
                }
            }

            public void PrintAllResultingPaths()
            {
                PrintAllResultingPaths(null);
            }

            public void PrintPartialPaths()
            {
                PrintAllResultingPaths(_graph.PartialPaths);
            }

            private void PrintAllResultingPaths(List<List<int>> myChosenPaths)
            {
                // Utilisation de l'opérateur de coalescence nulle (??) de C# :
                // Si myChosenPaths n'est pas null, on l'utilise, sinon on prend _graph.ResultingPaths
                List<List<int>> paths = myChosenPaths ?? _graph.ResultingPaths;

                if (paths != null)
                {
                    foreach (List<int> path in paths)
                    {
                        Console.WriteLine($"Path: [{string.Join(", ", path)}]");
                    }
                }
            }
        }
    }
}

File: examples/java/examples.graphe/src/main/java/org/clprolf/graph/workers/impl/GraphWorkerImpl.java
package org.clprolf.graph.workers.impl;

import org.clprolf.framework.ClWorker;
import org.clprolf.graph.agents.Graph;
import org.clprolf.graph.agents.impl.GraphImpl;
import org.clprolf.graph.workers.GraphWorker;

import java.util.List;
import java.util.Map;

@ClWorker
public class GraphWorkerImpl implements GraphWorker {
    private Graph graph;

    public GraphWorkerImpl(Graph theGraph){
        this.graph = theGraph;
    }

    // Display graph structure
    public void printGraph() {
        Map<Integer, List<Integer>> adj = graph.getAdj();
        for (var entry : adj.entrySet()) {
            System.out.println(entry.getKey() + " -> " + entry.getValue());
        }
    }

    public void printAllResultingPaths() {
        printAllResultingPaths(null);
    }

    public void printPartialPaths(){
        printAllResultingPaths(this.graph.getPartialPaths());
    }

    private void printAllResultingPaths(List<List<Integer>> myChosenPaths) {
        List<List<Integer>> paths;

        if (myChosenPaths != null) {
            paths = myChosenPaths;
        }
        else {
            paths = graph.getResultingPaths();
        }
        // Display paths separately
        for (List<Integer> path : paths) {
            System.out.println("Path: " + path);
        }
    }
}
